polling_stations/apps/data_finder/helpers/geocoders.py

```diff
@@ -150,10 +138,13 @@ def get_code(self, code_type, *args, **kwargs):
     raise PostcodeError("Could not geocode from any source")
 
 
-def get_council(geocode_result):
+def get_council(geocode_result, uprn=None):
     try:
-        return Council.objects.defer("area").get(
-            council_id=geocode_result.get_code("lad")
-        )
+        if uprn:
+            return Council.objects.defer("area").get(
+                council_id=geocode_result.get_code("lad", uprn=uprn)
+            )
+        else:
+            return Council.objects.defer("area").get(council_id=geocode_result.get_code("lad"))
```

Comment:
This feels a bit messy, but essentially we want to give a uprn to the `AddressbaseGeocoder`, but not to the `OnspdGeocoder`. We won't always have a uprn because we don't want to show everyone an address picker.